config_builder.go

```diff
@@ -915,6 +919,12 @@ func (d *DefaultDatabaseBuilder) BuildDatabase(
 	// using the same struct (and DB backend) instance.
 	dbs.ChanStateDB = dbs.GraphDB
 
+	// For now the only InvoiceDB implementation is the *channeldb.DB.
+	//
+	// TODO(positiveblue): use a sql first implementation for this
+	// interface.
+	dbs.InvoiceDB = dbs.GraphDB
```

Comment:
This should only set the pointer if it isn't already set. This way someone can instantiate the lnd daemon struct and pass in their own implementation of the interface.

Reply:
Can't they already do that providing a their own implementation of `DatabaseBuilder`?